homeassistant/components/dnsip/__init__.py

```diff
@@ -25,3 +26,37 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload dnsip config entry."""
 
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry to a newer version."""
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False  # pragma: no cover
```

Comment:
Note, I excluded this line from code coverage as it covers a hypothetical future scenario where a user downgrades from a future version to this version.
It is included in the first place as it exists in the example [developer documentation](https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#config-entry-migration)

Reply:
See https://github.com/home-assistant/core/blob/dev/tests/components/teslemetry/test_config_flow.py#L238 as example on how to test this.

Reply:
I will do this, but the reason I didn't was that in a future major version, by definition, both the code and the config entry will be of a significantly different structure, so we are not actually testing anything 'real', and the in that context a passing or failing test will not add any clarity.
But either way I will add the new test.

Reply:
Implemented in [ed7237d](https://github.com/home-assistant/core/pull/113993/commits/ed7237d31cd108607cc32e9b809616f8763f6be8)